pkg/logs/logs.go

```diff
@@ -0,0 +1,53 @@
+package logs
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/newrelic/newrelic-client-go/internal/http"
+	"github.com/newrelic/newrelic-client-go/internal/logging"
+	"github.com/newrelic/newrelic-client-go/pkg/config"
+)
+
+// Logs is used to send log data to the New Relic Log API
+type Logs struct {
+	client http.Client
+	config config.Config
+	logger logging.Logger
+}
+
+// New is used to create a new Logs client instance.
+func New(cfg config.Config) Logs {
+	cfg.Compression = config.Compression.Gzip
+
+	client := http.NewClient(cfg)
+	client.SetAuthStrategy(&http.LicenseKeyAuthorizer{})
+
+	pkg := Logs{
+		client: client,
+		config: cfg,
+		logger: cfg.GetLogger(),
+	}
+
+	return pkg
+}
+
+// CreateLogEntry reports a log entry to New Relic.
+// It's up to the caller to send a valid Log API payload, no checking done here
+func (l *Logs) CreateLogEntry(logEntry interface{}) error {
+	if logEntry == nil {
+		return errors.New("logs: CreateLogEntry: logEntry is nil, nothing to do")
+	}
+
+	rsp, err := l.client.Post(l.config.Region().LogsURL(), nil, logEntry, nil)
+
+	if err != nil {
+		return err
+	}
+
+	if !(rsp.StatusCode < 299) {
```

Comment:
+1 on @sanderblue 's comment from earlier.  This case should be covered from within the `Post` method, and so this check seems redundant.  By the time we get here, this conditional will always evaluate to true unless `rsp.StatusCode == 299`.

Reply:
For a reason I could not identify that code did not work, this was the only way to get it to succeed.

Reply:
I'm interested to dig a little deeper on this in case we have some bugs hidden in the client.  Do you have a repro case?  The integration test provided in this PR seems to succeed with or without this check.

That being said, we don't need to hold this PR up while I look into this.

Reply:
No, I don't. We'd have to rewrite that bit of code an try it out.